src/Input/Select/Select.tsx

```diff
@@ -162,6 +162,19 @@ const Select: ISelect = (props: Props) => {
     [value]
   );
 
+  React.useEffect(
+    function resetActiveOptionIndex() {
+      // When the nth item in the list is selected, the active index is n
+      // When the SelectList is opened again, there may be < n items if the
+      // list is filtered. For this case, we need to reset the active index
+      // to 0 to ensure that there is an 'active' item in the SelectList.
+      if (activeOptionIndex > options.length - 1) {
+        setActiveOptionIndex(0);
+      }
+    },
+    [activeOptionIndex, options.length]
+  );
```

Comment:
I think it will be safer to `findActiveOptionIndex` before update `activeOptionIndex`  instead of `setActiveOptionIndex(0)`.

And can consider handling this logic within current `updateOptions` `useEffect`.
```
  React.useEffect(
    function updateOptionsAndActiveIndex() {
      setOptions(availableOptions);

      const findActiveOptionIndex = availableOptions.findIndex(option =>
        toLower(option.props.children).includes(toLower(inputValue))
      );
      if (activeOptionIndex !== findActiveOptionIndex) {
        setActiveOptionIndex(findActiveOptionIndex);
      }
    },
    [availableOptions]
  );
```

Reply:
Seems like this will break some tests if I pass in the full dependency array `[activeOptionIndex, availableOptions, inputValue]`

Reply:
Okay it works! Just needed to add a condition. Thanks!